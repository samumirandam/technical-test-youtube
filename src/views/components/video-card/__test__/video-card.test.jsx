/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import VideoCard from '../index';

const defaultProps = {
  name: 'test image',
};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<VideoCard {...setupStore} />, {});
};

describe('Test for VideoCard component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByAltText('test image')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render click in button', () => {
    const props = {
      canDelete: true,
    };
    setup(props);
    fireEvent.click(screen.getByAltText('test image'));
    fireEvent.click(screen.getByTestId('VideoCard__delete'));
    expect(screen.getByAltText('test image')).toBeTruthy();
  });
});
