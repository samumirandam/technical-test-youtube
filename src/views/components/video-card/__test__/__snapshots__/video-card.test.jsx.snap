// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for VideoCard component Should renders the same component 1`] = `
<div>
  <div
    class="VideoCard"
  >
    <button
      class="Button noStyles"
      data-testid=""
      type="button"
    >
      <figure>
        <img
          alt="test image"
          src=""
        />
      </figure>
    </button>
  </div>
</div>
`;
